Python/ceval.c

```diff
@@ -4854,6 +4854,41 @@ _PyEval_EvalFrameDefault(PyThreadState *tstate, InterpreterFrame *frame, int thr
             goto start_frame;
         }
 
+        TARGET(CALL_NO_KW_TYPE_1) {
+            assert(STACK_ADJUST_IS_RESET);
+            assert(GET_CACHE()->adaptive.original_oparg == 1);
+            PyObject *obj = TOP();
+            PyObject *callable = SECOND();
+            DEOPT_IF(callable != (PyObject *)&PyType_Type, CALL_NO_KW);
```

Comment:
I don't remember if you like using the `Py_IS_TYPE` inline function, but just mentioning it here and leaving it up to you :).

Reply:
I'll skip on the obfuscation macro :slightly_smiling_face: